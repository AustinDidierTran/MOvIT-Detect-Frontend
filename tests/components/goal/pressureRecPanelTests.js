import React from 'react';
import PropTypes from 'prop-types';
import configureMockStore from 'redux-mock-store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';

import PressureRecPanel from '../../../src/components/goal/pressureRecPanel';

Enzyme.configure({ adapter: new Adapter() });

describe('PressureRecPanel Tests', () => {
  let wrapper = {};
  const initialState = {
    applicationReducer: { language: 'en' },
    recommendationReducer: { reduceWeight: true },
    configurationReducer: { maxAngle: 90 },
    goalReducer: {
      tiltFrequencyGoal: 900,
      tiltLengthGoal: 60,
      tiltAngleGoal: 40,
    },
  };
  const mockStore = configureMockStore();
  const store = mockStore(initialState);

  beforeEach(() => {
    wrapper = shallow(<PressureRecPanel store={store} />).dive();

    expect(wrapper.state('modifieGoal')).toEqual(false);
    expect(wrapper.state('maxSliderAngle')).toEqual(90);
  });


  it('should have proptypes', () => {
    // Actual value
    const actualValue = PressureRecPanel.WrappedComponent.propTypes;

    // Expected value
    const expectedValue = {
      language: PropTypes.string.isRequired,
      profile: PropTypes.string.isRequired,
      maxAngle: PropTypes.number,
      reduceWeight: PropTypes.bool.isRequired,
      tiltFrequencyGoal: PropTypes.number.isRequired,
      tiltLengthGoal: PropTypes.number.isRequired,
      tiltAngleGoal: PropTypes.number.isRequired,
      changeTiltFrequencyGoal: PropTypes.func.isRequired,
      changeTiltLengthGoal: PropTypes.func.isRequired,
      changeTiltAngleGoal: PropTypes.func.isRequired,
    };

    // Test
    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should display labels and not save anything when we\'re not modifying', () => {
    const spy = sinon.spy(wrapper.instance(), 'save');
    wrapper.instance().toggleEditing();

    expect(wrapper.state('modifieGoal')).toEqual(true);
    expect(spy.calledOnce).toEqual(false);
    expect(toJson(wrapper)).toMatchSnapshot(); // snapshot with sliders
  });

  it('should hide labels and save new data when we\'re modifying', () => {
    const spy = sinon.spy(wrapper.instance(), 'save');
    wrapper.setState({ modifieGoal: true });
    wrapper.instance().toggleEditing();

    expect(wrapper.state('modifieGoal')).toEqual(false);
    expect(spy.calledOnce).toEqual(true);
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
