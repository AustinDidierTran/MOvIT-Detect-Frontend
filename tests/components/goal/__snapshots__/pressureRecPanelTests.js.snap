// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressureRecPanel Tests should display labels and not save anything when we're not modifying 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <TiltSlidersCard
        id="reduceWeight"
        maxAngle={90}
        modifiable={false}
        onAngleChange={[Function]}
        onFrequencyChange={[Function]}
        onLengthChange={[Function]}
        onModifie={[Function]}
        tiltAngle={40}
        tiltFrequecy={900}
        tiltLength={60}
        title="recommendations.reduceWeight.en"
        tooltip="recommendations.reduceWeight.recPanel.tooltip.en"
      />
    </div>
  </div>
  <Tooltip
    appendTo="body"
    escape={true}
    for="#personalGoalInfo"
    hideDelay={null}
    onBeforeShow={null}
    positionStyle={null}
    showDelay={null}
    title="toolTip.personalGoal.en"
    tooltipDisabled={false}
    tooltipEvent="hover"
    tooltipPosition="right"
    tooltipStyleClass={null}
  />
</div>
`;

exports[`PressureRecPanel Tests should match the snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <Connect(TiltLabels)
        id="reduceWeight"
        modifiable={false}
        onModifie={[Function]}
        tiltAngle={40}
        tiltFrequecy={900}
        tiltLength={60}
        title="recommendations.reduceWeight.en"
        tooltip="recommendations.reduceWeight.recPanel.tooltip.en"
      />
    </div>
  </div>
  <Tooltip
    appendTo="body"
    escape={true}
    for="#personalGoalInfo"
    hideDelay={null}
    onBeforeShow={null}
    positionStyle={null}
    showDelay={null}
    title="toolTip.personalGoal.en"
    tooltipDisabled={false}
    tooltipEvent="hover"
    tooltipPosition="right"
    tooltipStyleClass={null}
  />
</div>
`;
